src/test/codegen/repr-u128.rs

```diff
@@ -0,0 +1,17 @@
+// compile-flags: --emit=llvm-ir -C debuginfo=2
+#![feature(repr128)]
+
+#[repr(u128)]
+pub enum Foo {
+    Lo,
+    Hi = 1 << 64,
+}
+
+pub fn foo() -> Option<Foo> {
+    None
+}
+
+// CHECK: declare void @llvm.dbg.value
```

Comment:
The part after CHECK is what we expect to find in the IR, in this case the `extraData: i128 ...` (I don't know the value right now) - although you should include the part of that line that indicates this is for the `None` variant, and put `{{.*}}` in between (the double braces are for regex patterns).

Also, it doesn't matter where you put the CHECK, relative to anything else in the file.
So it might be nice to put it between the `Foo` enum and the `foo` function.

Reply:
Hmm, I tested with `CHECK` like [this](https://github.com/rust-lang/rust/blob/fc8581d7420b63b2cc5eac1c57a40de785859448/src/test/codegen/enum-debug-niche.rs#L11-L22), but the test was failed.

Reply:
Did you have `// compile-flags: -g -C no-prepopulate-passes` at the top like the file you linked?

Also, what CHECK line did you use exactly?

Reply:
I added `compile-flags` you said and this CHECK  `// CHECK: {{.*}}DIDerivedType{{.*}}tag: DW_TAG_member,{{.*}}name: "Bar",{{.*}}extraData:{{.*}}`.
I think this CHECK might be wrong.

Reply:
You want "None", not "Bar", and look in the LLVM IR you get yourself, for this code, to get the extraData value.

Sorry for getting the compile flags wrong, the existing codegen tests are more correct.

Reply:
You basically need all of these lines, for this test to work as expected: https://github.com/rust-lang/rust/blob/fc8581d7420b63b2cc5eac1c57a40de785859448/src/test/codegen/enum-debug-niche.rs#L6-L9

Reply:
Thank you! I added WIP commit, Is my work correct so far?

Reply:
@eddyb Is this okay? I will work on codegen test at this moment, then debuginfo test.

Reply:
`// CHECK: declare void @llvm.dbg.value` doesn't need to exist, you only care about the `!DI` check.

For that check, you should put the expected value after `extraData` (you can get it out of the the `--emit=llvm-ir` output for this file, but I think it would be `18446745000000000124`).

Reply:
@eddyb Thank you! I fixed the checks of codegen test.